src/main/java/hillel/spring/reviews/ReviewService.java

```diff
@@ -0,0 +1,128 @@
+package hillel.spring.reviews;
+
+import hillel.spring.appointments.Appointment;
+import hillel.spring.appointments.AppointmentRepo;
+import hillel.spring.pet.PetRepo;
+import hillel.spring.pet.dto.PetNotFoundException;
+import hillel.spring.reviews.dto.ReviewDtoConverter;
+import hillel.spring.reviews.dto.ReviewInputWithTimeAndPetIdDto;
+import hillel.spring.reviews.dto.ReviewOutputDto;
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import lombok.val;
+import org.hibernate.StaleObjectStateException;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor
+public class ReviewService {
+    private final PetRepo petRepo;
+    private final ReviewRepo reviewRepo;
+    private final ReviewConfig reviewConfig;
+    private final AppointmentRepo appointmentRepo;
```

Comment:
it is better to avoid accessing other repositories from your service. Please rework this, and replace these repositories with appropriate services.